cmd/soroban-cli/src/config/locator.rs

```diff
@@ -0,0 +1,190 @@
+        self.network_dir().map(|p| {
+            let mut source = p.join(name);
+            source.set_extension("toml");
+            source
+        })
+    }
+
+    pub fn write_identity(&self, name: &str, secret: &Secret) -> Result<(), Error> {
+        let source = self.identity_path(name)?;
+        let data = toml::to_string(secret).map_err(|_| Error::ConfigSerialization)?;
+        std::fs::write(&source, data).map_err(|error| Error::IdCreationFailed {
+            filepath: source.clone(),
+            error,
+        })
+    }
+
+    pub fn write_network(&self, name: &str, network: &Network) -> Result<(), Error> {
+        let source = self.network_path(name)?;
+        let data = toml::to_string(network).map_err(|_| Error::Deserialization)?;
+        std::fs::write(source, data).map_err(|_| Error::NetworkCreationFailed)
+    }
+
+    pub fn list_identities(&self) -> Result<Vec<String>, Error> {
+        let path = self.identity_dir()?;
+        read_dir(&path)
+    }
+
+    pub fn list_networks(&self) -> Result<Vec<String>, Error> {
+        let path = self.network_dir()?;
+        read_dir(&path)
+    }
+
+    // pub fn config_path(&self) -> Result<PathBuf, Error> {
+    //     Ok(self.config_dir()?.join("config.toml"))
+    // }
+
+    // pub fn get_config_file(&self) -> Result<Config, Error> {
+    //     let path = self.config_path()?;
+    //     if path.exists() {
+    //         let data = fs::read(&path).map_err(|_| Error::CannotReadConfigFile)?;
+    //         toml::from_slice::<Config>(&data).map_err(|_| Error::Deserialization)
+    //     } else {
+    //         Ok(Config::default())
+    //     }
+    // }
+
+    // pub fn write_config_file(&self, config: &Config) -> Result<(), Error> {
+    //     let path = self.config_path()?;
+    //     let data = toml::to_string(config).map_err(|_| Error::ConfigSerialization)?;
+    //     fs::write(path, data).map_err(|_| Error::CannotWriteConfigFile)
+    // }
+}
+
+pub fn read_identity(name: &str) -> Result<Secret, Error> {
+    // 1. check workspace config files for `name`
+    let local_identity = Args { global: false }.identity_path(name);
+    // 2. use if found, else, check global config files for `name`
+    let path = local_identity.or_else(|_| Args { global: true }.identity_path(name))?;
+
+    println!("Found identity \"{name}\" at {}", path.display());
+
+    let data = fs::read(&path).map_err(|_| Error::SecretFileRead { path })?;
+    toml::from_slice::<Secret>(&data).map_err(|_| Error::Deserialization)
+}
+
+pub fn read_network(name: &str) -> Result<Network, Error> {
+    // 1. check workspace config files for `name`
+    let local_network = Args { global: false }.network_path(name);
+    // 2. use if found, else, check global config files for `name`
+    let path = local_network.or_else(|_| Args { global: true }.identity_path(name))?;
+
+    println!("Found network \"{name}\" at {}", path.display());
```

Comment:
Should we maybe use `eprintln!`? or remove the printlns to avoid polluting the output of other commands?